### PR TITLE
Create toArray.md

### LINKED ISSUE #4105: [Term Entry] Java ArrayList: .toArray()
### Reason/inspiration (optional)

We want to add the `.toArray()` method as a new term entry for [ArrayList](https://www.codecademy.com/resources/docs/java/array-list). Provide the entry in a markdown file and place it under the file path `content/java/concepts/array-list/terms/toArray/toArray.md`

The entry should have the following:
– An introductory paragraph about what the method does.
– A syntax section with a pseudo code block explaining it.
– An example showing the method in use.

Please refer to the [term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Brag2gr8**
### Description

Added new entry for Javascript Object Methods defineProperties()
Entry Location: content/javascript/concepts/objects/terms/toArray/toArray.md. closes #4105

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/java/concepts/array-list/terms/toArray/toArray.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+---
+Title: '.toArray()'
+Description: 'Converts a collection or iterable object into an array.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Arrays'
+  - 'Data Types'
```
Comment (Sriparno08) [inline content/java/concepts/array-list/terms/toArray/toArray.md:9]: Add `'Elements'` inside the `Tags` section:

```suggestion
Tags:
  - 'Arrays'
  - 'Data Types'
  - 'Elements'
```

**Review by Sriparno08 — COMMENTED** [review]
Hey @Brag2gr8, thanks for contributing to Codecademy!

I have listed all the modifications that you need to make in your article below:

**Comment by Brag2gr8** [comment]
@Sriparno08 I just finished comiting the changes

**Comment by Sriparno08** [comment]
Good job, @Brag2gr8 👍

This looks good for a second review.

File: content/java/concepts/array-list/terms/toArray/toArray.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+public class Main {
+    public static void main(String[] args) {
+        ArrayList<String> fruitsList = new ArrayList<>();
+        fruitsList.add("Apple");
+        fruitsList.add("Banana");
+        fruitsList.add("Orange");
+
+        // Converting the ArrayList to an array of strings
+        String[] fruitsArray = fruitsList.toArray(new String[0]);
+
```
Comment (avdhoottt) [inline content/java/concepts/array-list/terms/toArray/toArray.md:51]:
```suggestion
```

File: content/java/concepts/array-list/terms/toArray/toArray.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+public class Main {
+    public static void main(String[] args) {
+        ArrayList<String> colorsList = new ArrayList<>();
+        colorsList.add("Red");
+        colorsList.add("Green");
+        colorsList.add("Blue");
+
+        // Converting the ArrayList to an array of objects
+        Object[] colorsArray = colorsList.toArray();
+
```
Comment (avdhoottt) [inline content/java/concepts/array-list/terms/toArray/toArray.md:81]:
```suggestion
```

File: content/java/concepts/array-list/terms/toArray/toArray.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        // Converting the ArrayList to an array of strings
+        String[] fruitsArray = fruitsList.toArray(new String[0]);
+
+        System.out.println("Fruits Array: " + Arrays.toString(fruitsArray));
+    }
+}
+```
+
+The output should look like this:
```
Comment (avdhoottt) [inline content/java/concepts/array-list/terms/toArray/toArray.md:57]:
```suggestion
The output should result in the following output:
```

File: content/java/concepts/array-list/terms/toArray/toArray.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+        // Converting the ArrayList to an array of objects
+        Object[] colorsArray = colorsList.toArray();
+
+        System.out.println("Colors Array: " + Arrays.toString(colorsArray));
+    }
+}
+```
+
+The output should look like this:
```
Comment (avdhoottt) [inline content/java/concepts/array-list/terms/toArray/toArray.md:87]:
```suggestion
The output should result in the following output:
```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@Brag2gr8 Suggested some minor changes, please make them.

**Comment by Brag2gr8** [comment]
@avdhoottt I'm done making the suggested changes

**Pull request merged**